probe-rs/src/probe/stlink/tools.rs

```diff
@@ -71,6 +71,10 @@ pub(super) fn read_serial_number<T: rusb::UsbContext>(
     let timeout = Duration::from_millis(100);
 
     let handle = device.open()?;
-    let language = handle.read_languages(timeout)?[0];
+    let language = handle
+        .read_languages(timeout)?
+        .get(0)
+        .cloned()
+        .ok_or(rusb::Error::BadDescriptor)?;
```

Comment:
This is indeed kinda hacky as Adam pointed out, but I believe it is fair enough, what do you think @adamgreig ?

Reply:
I'd like it to do whatever it does for devices without a serial number (which a valid cmsis-dap probe might have), but I'm not sure what it does in that case... I will try and find out now.

Reply:
Ok, such devices return `InvalidParam` which isn't really any different, so I don't mind what we do here, this seems as good as anything.

Reply:
Thanks for checking!